### ISSUE TITLE
make error - libgnuradio-dsdcc.dylib 

### ISSUE THREAD

**Issue by rkarikari** [argilo/gr-dsdcc#2]
Hello,

I get the following error when trying to build....

<img width="677" alt="Screenshot 2023-11-25 at 6 04 05 PM" src="https://github.com/argilo/gr-dsdcc/assets/115777368/494d8a3d-a2cf-4eff-a867-b8464706fca6">

.
.
.
.

The file  " libgnuradio-dsdcc.dylib " does not exist and is not created.

Any suggestions / help would be greatly appreciated.

**Comment by rkarikari** [comment]
`
➜  build git:(main) cmake ../
-- The CXX compiler identification is AppleClang 12.0.0.12000031
-- The C compiler identification is AppleClang 12.0.0.12000031
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /Library/Developer/CommandLineTools/usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /Library/Developer/CommandLineTools/usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE  
-- Found PkgConfig: /opt/local/bin/pkg-config (found version "0.29.2") 
-- Checking for module 'gmp'
--   Found gmp, version 6.3.0
-- Found GMP: /usr/local/Cellar/gmp/6.3.0/lib/libgmpxx.dylib  
-- Using GMP.
-- Found MPLIB: /usr/local/Cellar/gmp/6.3.0/lib/libgmpxx.dylib  
-- Found Boost: /usr/local/lib/cmake/Boost-1.83.0/BoostConfig.cmake (found suitable version "1.83.0", minimum required is "1.83.0") found components: date_time program_options system regex thread 
-- Found Volk: Volk::volk  
-- User set python executable /usr/local/Cellar/gnuradio/3.10.7.0_3/libexec/venv/bin/python
CMake Warning (dev) at /usr/local/lib/cmake/gnuradio/GrPython.cmake:21 (find_package):
  Policy CMP0148 is not set: The FindPythonInterp and FindPythonLibs modules
  are removed.  Run "cmake --help-policy CMP0148" for policy details.  Use
  the cmake_policy command to set the policy and suppress this warning.

Call Stack (most recent call first):
  /usr/local/lib/cmake/gnuradio/GnuradioConfig.cmake:37 (include)
  CMakeLists.txt:30 (find_package)
This warning is for project developers.  Use -Wno-dev to suppress it.

-- Found PythonInterp: /usr/local/Cellar/gnuradio/3.10.7.0_3/libexec/venv/bin/python (found version "3.11.6") 
CMake Warning (dev) at /usr/local/lib/cmake/gnuradio/GrPython.cmake:27 (find_package):
  Policy CMP0148 is not set: The FindPythonInterp and FindPythonLibs modules
  are removed.  Run "cmake --help-policy CMP0148" for policy details.  Use
  the cmake_policy command to set the policy and suppress this warning.

Call Stack (most recent call first):
  /usr/local/lib/cmake/gnuradio/GnuradioConfig.cmake:37 (include)
  CMakeLists.txt:30 (find_package)
This warning is for project developers.  Use -Wno-dev to suppress it.

-- Found PythonLibs: /usr/local/Frameworks/Python.framework/Versions/3.11/lib/libpython3.11.dylib (found suitable exact version "3.11.6") 
-- Checking for module 'libdsdcc'
--   Found libdsdcc, version 1.9.0
-- Performing Test HAVE_VISIBILITY_HIDDEN
-- Performing Test HAVE_VISIBILITY_HIDDEN - Success
-- Performing Test HAVE_WARN_SIGN_COMPARE
-- Performing Test HAVE_WARN_SIGN_COMPARE - Success
-- Performing Test HAVE_WARN_ALL
-- Performing Test HAVE_WARN_ALL - Success
-- Performing Test HAVE_WARN_NO_UNINITIALIZED
-- Performing Test HAVE_WARN_NO_UNINITIALIZED - Success
-- Found Git: /usr/bin/git  
-- Extracting version information from git describe...
fatal: No names found, cannot describe anything.
-- Found Doxygen: /opt/local/bin/doxygen (found version "1.9.8") found components: doxygen dot 
-- Using install prefix: /usr/local
-- Building for version: 1.0.0.0 / 1.0.0
-- No C++ unit tests... skipping
-- PYTHON and GRC components are enabled
-- Python checking for pygccxml - found
CMake Warning (dev) at /usr/local/share/cmake/pybind11/FindPythonLibsNew.cmake:98 (find_package):
  Policy CMP0148 is not set: The FindPythonInterp and FindPythonLibs modules
  are removed.  Run "cmake --help-policy CMP0148" for policy details.  Use
  the cmake_policy command to set the policy and suppress this warning.

Call Stack (most recent call first):
  /usr/local/share/cmake/pybind11/pybind11Tools.cmake:50 (find_package)
  /usr/local/share/cmake/pybind11/pybind11Common.cmake:188 (include)
  /usr/local/share/cmake/pybind11/pybind11Config.cmake:250 (include)
  /usr/local/lib/cmake/gnuradio/GrPybind.cmake:3 (find_package)
  python/dsdcc/bindings/CMakeLists.txt:25 (include)
This warning is for project developers.  Use -Wno-dev to suppress it.

-- Found PythonInterp: /usr/local/Cellar/gnuradio/3.10.7.0_3/libexec/venv/bin/python  
-- Found PythonLibs: /usr/local/Frameworks/Python.framework/Versions/3.11/lib/libpython3.11.dylib
-- Performing Test HAS_FLTO
-- Performing Test HAS_FLTO - Success
-- Performing Test HAS_FLTO_THIN
-- Performing Test HAS_FLTO_THIN - Success
-- Found pybind11: /usr/local/include (found version "2.11.1")
-- Configuring done (4.1s)
CMake Warning (dev):
  Policy CMP0068 is not set: RPATH settings on macOS do not affect
  install_name.  Run "cmake --help-policy CMP0068" for policy details.  Use
  the cmake_policy command to set the policy and suppress this warning.

  For compatibility with older versions of CMake, the install_name fields for
  the following targets are still affected by RPATH settings:

   gnuradio-dsdcc

This warning is for project developers.  Use -Wno-dev to suppress it.

-- Generating done (0.1s)

`

**Comment by rkarikari** [comment]
<img width="311" alt="Screenshot 2023-11-25 at 6 14 18 PM" src="https://github.com/argilo/gr-dsdcc/assets/115777368/bee29223-e0ce-4d7a-a054-ccbca01ea0e8">

**Comment by argilo** [comment]
It would appear that you don't have DSDcc installed.

**Comment by rkarikari** [comment]
Thanks for the quick response.

I do have DSDcc installed...

<img width="978" alt="Screenshot 2023-11-25 at 7 23 04 PM" src="https://github.com/argilo/gr-dsdcc/assets/115777368/f8d26cf8-f631-4df0-aae9-d8584d861d93">

**Comment by argilo** [comment]
OK. Then it looks like for some reason gr-dsdcc isn't able to find it. Unfortunately I don't have any more suggestions so you'll have to investigate on your own. Please let me know if you find a solution.

**Comment by rkarikari** [comment]
comparing .....  

     gr-dsdcc 
to
   `  gr-dsd    `            
 (which I have installed and works great),

there appears to be an entire folder/sub-directory "dsdcc"    missing !

no source files !

**Comment by argilo** [comment]
> there appears to be an entire folder/sub-directory "dsdcc" missing

It is not missing. The gr-dsd project includes a copy of dsd because it needs to make some special modifications to it. But gr-dsdcc simply uses the unmodified dsdcc library, which must already be installed on the system.

**Comment by rkarikari** [comment]
I got it to install with the following additions ...

```
export PKG_CONFIG_PATH=$PKG_CONFIG_PATH:/usr/local/lib/pkgconfig:/usr/local/lib 

export LIBRARY_PATH=$LIBRARY_PATH:/usr/local/lib
```
... to  my .zshrc file


However, it does not decode.
This attached file works with the gr-dsd block in place of  the gr-dsdcc.

[DSDcc.grc.zip](https://github.com/argilo/gr-dsdcc/files/13466941/DSDcc.grc.zip)

**Comment by argilo** [comment]
Does the stand-alone DSDcc application decode? It's a different decoder from DSD, so results may vary. Configuration may also be different.

**Comment by rkarikari** [comment]
I finally got everything working!

DSDCC  installed from source (github) did not work for some reason .
not sure why .... perhaps I messed something up during  my troubleshooting.

I manually uninstalled it completely ( reversing all changes made by ' make install '   ).

I then installed dsdcc via macports ( same version )

Tested it to confirm it decodes .

Then re-build gr-dsdcc  to use the new files  .. (this time the libraries were installed in.... )

` /opt/local/lib `

instead of 

` /usr/local/lib `

In GnuRadio companion,  the dsdcc block works now .

Thanks so much for the suggestions.
